common/djangoapps/student/views.py

```diff
@@ -520,6 +515,13 @@ def dashboard(request):
         course_enrollment_pairs, course_modes_by_course
     )
 
+    # Retrieve the course modes for each course
+    enrolled_courses_dict = {}
+    for course, __ in course_enrollment_pairs:
+        enrolled_courses_dict[unicode(course.id)] = course
+
+    credit_messages = _create_credit_availability_message(enrolled_courses_dict, user)
```

Comment:
It executes 2 queries per enrollment 1 for getting the user eligibility in the course and 1 for getting the credit request against the given course if student has made any. I have added a test for this.